### ISSUE TITLE
Warning, when using "usort" with mocks

### ISSUE THREAD

**Issue by aik099** [phpspec/prophecy#215]
Consider following code:
```php
$object1 = $this->prophesize('SomeNS\\SomeClasss');
$object1->getWeight()->willReturn(2);

$object2 = $this->prophesize('SomeNS\\SomeClasss');
$object2->getWeight()->willReturn(1);

$objects = array($object1, $object2);

usort($objects, function ($a, $b) {
    if ( $a->getWeight() == $b->getWeight() ) {
        return 0;
    }

    return ($a->getWeight() < $b->getWeight()) ? -1 : 1;
});
```
This results in sorting not working and following notice/warning shown:
```
usort(): Array was modified by the user comparison function
```

This is known PHP bug (see https://bugs.php.net/bug.php?id=50688), but I'm not changing/inspecting array inside the sorting function. I bet Prophecy does in some way and that cases the problem.

**Comment by aik099** [comment]
Maybe what's happening is each mock recording made calls while being used from comparison function and that makes `usort` unhappy.

**Comment by mbrodala** [comment]
Aren't you missing two `->reveal()` calls here? Right now you are sorting `ObjectProphecy` instances instead of `SomeClasss` instances.

**Comment by aik099** [comment]
Luckily the `ObjectProphecy` class has `__call` method that works the same way as `__call` method on the reveal prophecy objects. That's why it works as needed. Adding `->reveal()` won't change much.

**Comment by mbrodala** [comment]
Didn't know that `SomeClasss` has a `_call()` method. ;-)

**Comment by aik099** [comment]
> the same way as `__call` method on the revealed prophecy objects

Maybe I'm not exactly correct and revealed object just have all their methods overridden to verify expectations.

**Comment by mbrodala** [comment]
Yes, that's the case. Prophecy builds a deriving class at runtime which implements all methods separately.
